### ISSUE TITLE
PeptideShaker did not manage to map most peptides to the database. Please verify your database.

### ISSUE THREAD

**Issue by VladimirRoudko** [compomics/peptide-shaker#408]
Hello,

I am running command-line version of peptide-shaker (Ubuntu, version 1.16.45) and getting the following STDOUT upon analysis of search results (done by command-line searchGUI, version 3.3.20, Ubuntu build):

**Mon Jun 22 15:44:55 EDT 2020 Warning: More than 75% of the PSMs did not pass the import filters.
 PeptideShaker did not manage to map most peptides to the database. Please verify your database.
 Apparently your database contains a high degree of shared peptides between the target and decoy sequences. Please verify your database.
 Please verify that your peptide selection criteria are not too restrictive.**

I use translated RefSeq database including decoys (reverse sequences). What could be the reason for such warning and should I worry about the final results?

Thank you,
Vladimir

**Comment by hbarsnes** [comment]
Hi Vladimir,

Peptides mapping to both target and decoy protein sequences are currently kicked out. Please ensure that the decoys were added correctly. Was this done in SearchGUI? Then make sure that you are using the same database in SearchGUI when doing the searches as when loading the search results in PeptideShaker. Finally, check that the database is big enough, and contains the correct protein sequences.

Best regards,
Harald

**Comment by VladimirRoudko** [comment]
Hi Harald,

Thank you for the suggestions! I indeed generate the decoys with SearchGUI and use the same copy of the generated reference for search and PeptideShaker. I noted, that this warning comes always after processing novor search results - any other search engine does not produce this message. Could it be something tool-specific? Could it be, I need to tune novor parameters better?

Thank you,
Vladimir

**Comment by VladimirRoudko** [comment]
Just in case it helps, here is the PeptideShaker part of STDOUT related to Novor search processing:

Tue Jun 23 11:06:55 EDT 2020 Unzipping searchgui_out.zip.
10% 20% 30% 40% 50% 60% 70% 80% 90% 100%

Tue Jun 23 11:07:41 EDT 2020 Import process for myExperiment (Sample: 04CPTAC_COprospective_W_VU_20150915_05CO006, Replicate: 1)

Tue Jun 23 11:07:41 EDT 2020 Importing sequences from human.ENSP_concatenated_target_decoy.fasta.
 0%
10% 20% 30% 40% 50% 60% 70% 80% 90% 100%
Tue Jun 23 11:08:23 EDT 2020 FASTA file import completed.
Tue Jun 23 11:08:23 EDT 2020 Importing gene mappings.
Tue Jun 23 11:08:25 EDT 2020 Establishing local database connection.
Tue Jun 23 11:08:26 EDT 2020 Reading identification files.

...................

Tue Jun 23 11:08:43 EDT 2020 Loading spectra for 04CPTAC_COprospective_W_VU_20150915_05CO006_f01.novor.csv.
10% 20% 30% 40% 50% 60% 70% 80% 90% 100%
Tue Jun 23 11:08:43 EDT 2020 Collecting peptides to map.
10% 20% 30% 40% 50% 60% 70% 80% 90% 100%
Tue Jun 23 11:08:43 EDT 2020 Mapping peptides to proteins.
10% 20% 30% 40% 50% 60% 70% 80% 90% 100%
Tue Jun 23 11:08:44 EDT 2020 Importing PSMs from 04CPTAC_COprospective_W_VU_20150915_05CO006_f01.novor.csv
10% 20% 30% 40% 50% 60% 70% 80%
Tue Jun 23 11:08:44 EDT 2020 19548 identified spectra (75.3%) did not present a valid peptide.
Tue Jun 23 11:08:44 EDT 2020 23314 of the best scoring peptides were excluded by the import filters:
Tue Jun 23 11:08:44 EDT 2020     - 91.8% peptide not matching to the database.
Tue Jun 23 11:08:44 EDT 2020     - 7.9% peptide length less than 7 or greater than 30.
Tue Jun 23 11:08:44 EDT 2020 Warning: More than 75% of the PSMs did not pass the import filters.
 PeptideShaker did not manage to map most peptides to the database. Please verify your database.
 Apparently your database contains a high degree of shared peptides between the target and decoy sequences. Please verify your database.
 Please verify that your peptide selection criteria are not too restrictive.

.............

Tue Jun 23 11:09:12 EDT 2020 Importing PSMs from 04CPTAC_COprospective_W_VU_20150915_05CO006_f02.novor.csv
10% 20% 30% 40% 50% 60% 70% 80% 90%
Tue Jun 23 11:09:12 EDT 2020 22994 identified spectra (85.2%) did not present a valid peptide.
Tue Jun 23 11:09:12 EDT 2020 24931 of the best scoring peptides were excluded by the import filters:
Tue Jun 23 11:09:12 EDT 2020     - 91.1% peptide not matching to the database.
Tue Jun 23 11:09:12 EDT 2020     - 8.6% peptide length less than 7 or greater than 30.
Tue Jun 23 11:09:12 EDT 2020 Warning: More than 75% of the PSMs did not pass the import filters.
 PeptideShaker did not manage to map most peptides to the database. Please verify your database.
 Apparently your database contains a high degree of shared peptides between the target and decoy sequences. Please verify your database.
 Please verify that your peptide selection criteria are not too restrictive.

...........

Tue Jun 23 11:09:40 EDT 2020 Loading spectra for 04CPTAC_COprospective_W_VU_20150915_05CO006_f03.novor.csv.
10% 20% 30% 40% 50% 60% 70% 80% 90% 100%
Tue Jun 23 11:09:40 EDT 2020 Collecting peptides to map.
10% 20% 30% 40% 50% 60% 70% 80% 90% 100%
Tue Jun 23 11:09:40 EDT 2020 Mapping peptides to proteins.
10% 20% 30% 40% 50% 60% 70% 80% 90% 100%
Tue Jun 23 11:09:40 EDT 2020 Importing PSMs from 04CPTAC_COprospective_W_VU_20150915_05CO006_f03.novor.csv
10% 20% 30% 40% 50% 60% 70% 80% 90%
Tue Jun 23 11:09:40 EDT 2020 24877 identified spectra (85.3%) did not present a valid peptide.
Tue Jun 23 11:09:40 EDT 2020 26869 of the best scoring peptides were excluded by the import filters:
Tue Jun 23 11:09:40 EDT 2020     - 91.8% peptide not matching to the database.
Tue Jun 23 11:09:40 EDT 2020     - 7.9% peptide length less than 7 or greater than 30.
Tue Jun 23 11:09:40 EDT 2020 Warning: More than 75% of the PSMs did not pass the import filters.
 PeptideShaker did not manage to map most peptides to the database. Please verify your database.
 Apparently your database contains a high degree of shared peptides between the target and decoy sequences. Please verify your database.
 Please verify that your peptide selection criteria are not too restrictive.

.....

Basically, each STDOUT related to Novor search results is exactly the same. Other tools do not produce the same warning message.  

thank you again,
Vladimir

**Comment by hbarsnes** [comment]
Hi Vladimir,

> Basically, each STDOUT related to Novor search results is exactly the same. Other tools do not produce the same warning message.

Ah, I did not notice that it was Novor-related. That explains a lot. The thing is that Novor uses a de novo based approach and thus does not use the protein sequences database when performing the peptide to protein matching. In fact, it does not even match peptides to the spectra but rather shorter tags, which it then fills out to make complete peptide sequences. It is therefore not surprising when many of these "artificial" peptides do not match any of the potential peptides from the protein sequence database.

DirecTag uses a similar approach, but only returns the shorter amino acid sequence tags along with mass gaps before/after the tags. This allows for more room when trying to map the search results onto the protein sequences and in many cases results in a higher percentage of mapped tags/peptides.

Generally, if using either of these two de novo engines (and a sequence database is available for your experimental setup) I'd recommend also including at least one of the "normal" database search engine as well. That way, the database search engine will find most of the common peptides, while the de novo engines (potentially) can add some of the less common peptides that would otherwise not be picked up. Whether this actually makes any difference (compared to just using "normal" search engines) very much depends on the dataset and the database though.

Best regards,
Harald

**Comment by VladimirRoudko** [comment]
Hi Harald,

Oh, I see! I assume then, that's warning message as a correct functioning of the Novor search. And I will definitely combine the output of de-novo and normal database search engines to get the most robust results. 
I wonder did anyone perform a kind of benchmarking of all available search engines separately on the same test MS/MS dataset? Just out of curiosity, what is the concordance between different tools using the same reference (let's say human proteome). 
Sorry for potentially extremely obvious question, I am just very novice to the proteomic field of research and trying to familiarize with the MS/MS analysis specifics. Also, if you could suggest any helpful readings I would be very grateful!

Thank you,
Vladimir

**Comment by hbarsnes** [comment]
Hi Vladimir,

> I wonder did anyone perform a kind of benchmarking of all available search engines separately on the same test MS/MS dataset?

There are several papers comparing different search engines, but the results often differ based on the dataset used. Such studies are also very much depending on being able to finetune all of the search engines. Comparisons therefore tend to suffer because one search engine, most often the one made by the authors of the paper, is better tuned to the dataset in question. Not necessarily on purpose, but mainly because the authors will always be most familiar with their own search engine.

For a comparison of the search engines in SearchGUI you can check out the Spectrum IDs tab in PeptideShaker, which shows you how many PSMs that were detected, both in total and unique to each algorithm. Usually the overlapp is quite high. Going beyond 2-3 search engines is therefore in most cases not worth the extra time it will take to process the data.

A tip would be to process a smaller test file with all of the search engine, and use these results as a guide for which search engine combination that seems to work best for that particular dataset.

> Also, if you could suggest any helpful readings I would be very grateful!

If you haven't already, I'd highly recommend taking a look at our tutorial material: http://compomics.com/bioinformatics-for-proteomics. It also includes links to more reading material.

Best regards,
Harald

**Comment by VladimirRoudko** [comment]
Great, thank you for the explanation and the reference! I indeed studied the tutorial, but will have a look into the references.

Thank you and closing the issue then.
Vladimir
